eth/tracers/internal/tracetest/supply_test.go

```diff
@@ -0,0 +1,695 @@
+	}
+
+	// Check balance at state:
+	// 1. A has 0 ether
+	// 2. B has 0 ether
+	statedb, _ := preCancunChain.State()
+	if got, exp := statedb.GetBalance(aa), big.NewInt(0); got.Cmp(exp) != 0 {
+		t.Fatalf("Pre-cancun address \"%v\" balance, got %v exp %v\n", aa, got, exp)
+	}
+	if got, exp := statedb.GetBalance(bb), big.NewInt(0); got.Cmp(exp) != 0 {
+		t.Fatalf("Pre-cancun address \"%v\" balance, got %v exp %v\n", bb, got, exp)
+	}
+
+	// Check live trace output
+	expected := live.SupplyInfo{
+		Delta:       big.NewInt(-31144500000000),
+		Reward:      common.Big0,
+		Withdrawals: common.Big0,
+		Burn:        big.NewInt(31144500000000),
+		Number:      1,
+		Hash:        common.HexToHash("0x71aafcce72971d4288bc4e50ca67aa0c6157cf9c1e135701f5b90c545bfaa136"),
+		ParentHash:  common.HexToHash("0x69cf2998c223bb2ad5a67c1f1c7a65cdc6ca4e744993897ea9b00dff4bb0e100"),
+	}
+
+	actual := preCancunOutput[expected.Number]
+
+	if !reflect.DeepEqual(actual, expected) {
+		t.Fatalf("Pre-cancun incorrect supply info: expected %+v, got %+v", expected, actual)
+	}
+
+	// 2. Test post Cancun
+	cancunTime := uint64(0)
+	gspec.Config.ShanghaiTime = &cancunTime
+	gspec.Config.CancunTime = &cancunTime
+
+	postCancunOutput, postCancunChain, err := testSupplyTracer(gspec, testBlockGenerationFunc)
+	if err != nil {
+		t.Fatalf("Post-cancun failed to test supply tracer: %v", err)
+	}
+
+	// Check balance at state:
+	// 1. A has 0 ether
+	// 2. B has 1 ether and 5 gwei
+	statedb, _ = postCancunChain.State()
+	if got, exp := statedb.GetBalance(aa), big.NewInt(0); got.Cmp(exp) != 0 {
+		t.Fatalf("Post-shanghai address \"%v\" balance, got %v exp %v\n", aa, got, exp)
+	}
+	if got, exp := statedb.GetBalance(bb), big.NewInt(0).Add(funds, gwei5); got.Cmp(exp) != 0 {
+		t.Fatalf("Post-shanghai address \"%v\" balance, got %v exp %v\n", bb, got, exp)
+	}
+
+	// Check live trace output
+	expected = live.SupplyInfo{
+		Delta:       big.NewInt(-31139500000000),
+		Reward:      common.Big0,
+		Withdrawals: common.Big0,
+		Burn:        big.NewInt(31139500000000),
+		Number:      1,
+		Hash:        common.HexToHash("0xca887ca0d205de78203355699fc2207696cb9d1a3cf2606abf43094deee95724"),
+		ParentHash:  common.HexToHash("0xf18527496dc26564d28a69ee3cfbc81877f7c1dc80f25d111cd9c097048442d3"),
+	}
+
+	actual = postCancunOutput[expected.Number]
+
+	if !reflect.DeepEqual(actual, expected) {
+		t.Fatalf("Post-cancun incorrect supply info: expected %+v, got %+v", expected, actual)
+	}
+}
+
+// Tests selfdestructing contract to send its balance to itself (burn).
+// It tests both cases of selfdestructing succeding and being reverted.
+//   - Contract A calls B and D.
+//   - Contract B selfdestructs and sends the funds to itself (Burn amount to be counted).
+//   - Contract C selfdestructs and sends the funds to itself.
+//   - Contract D calls C and reverts (Burn amount of C
+//     has to be reverted as well).
+func TestSupplySelfdestructItselfAndRevert(t *testing.T) {
```

Comment:
@s1na this tests handles both cases of counting burned amount when selfdestruct occurs + when it reverts.
For this reason I will remove the previous test.

Reply:
Remove the referenced test with [694c51d](https://github.com/s1na/go-ethereum/pull/21/commits/694c51de943d7cf3c1a8927496a06da765b95daa)